src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clones = [Object.assign({}, state)];
+
+  actions.forEach((action) => {
+    const { extraData } = action;
```

Comment:
Consider it will be better
```suggestion
    const { extraData, type, keysToRemove } = action;
```